website/utils/test/url_utils.test.ts

```diff
@@ -0,0 +1,272 @@
+        color: "ff0000",
+        name: undefined,
+        match: 1,
+        colorizeEnabled: true,
+        colorizeAlpha: 0.5,
+        surfaceOpacity: 0.75,
+        surfaceEnabled: false,
+        enabled: false,
+        isovalue: 0,
+        lut: ["0", "255"],
+      },
+    ],
+    [
+      "c5=v:0,c:00ff00,cz:0,cza:0,op:1,lut:autoij:,i:1,iv:100",
+      "c5=v%3A0%2Cc%3A00ff00%2Ccz%3A0%2Ccza%3A0%2Cop%3A1%2Clut%3Aautoij%3A%2Ci%3A1%2Civ%3A100",
+      {
+        color: "00ff00",
+        name: undefined,
+        match: 5,
+        colorizeEnabled: false,
+        colorizeAlpha: 0,
+        surfaceOpacity: 1,
+        surfaceEnabled: true,
+        enabled: false,
+        isovalue: 100,
+        lut: ["autoij", ""],
+      },
+    ],
+  ];
+
+  it("parses unencoded per-channel setting", async () => {
+    for (const [queryString, , expected] of channelParamToSetting) {
+      const params = new URLSearchParams(queryString);
+      const { args } = await getArgsFromParams(params);
+      const channelSetting = args.viewerChannelSettings?.groups[0].channels[0]!;
+      expect(channelSetting).toEqual(expected);
+    }
+  });
+
+  it("parses encoded per-channel settings", async () => {
+    for (const [, queryString, expected] of channelParamToSetting) {
+      const params = new URLSearchParams(queryString);
+      const { args } = await getArgsFromParams(params);
+      const channelSetting = args.viewerChannelSettings?.groups[0].channels[0]!;
+      expect(channelSetting).toEqual(expected);
+    }
+  });
+
+  it("parses multiple per-channel settings", async () => {
+    // Test unencoded (i=0) and encoded (i=1)
+    for (let i = 0; i < 2; i++) {
+      const queryString =
+        channelParamToSetting[0][i] + "&" + channelParamToSetting[1][i] + "&" + channelParamToSetting[2][i];
+      const params = new URLSearchParams(queryString);
+      const { args } = await getArgsFromParams(params);
+      const channelSettings = args.viewerChannelSettings?.groups[0].channels!;
+
+      // Order is not guaranteed, so check if any of the expected settings are present
+      expect(channelSettings).toContainEqual(channelParamToSetting[0][2]);
+      expect(channelSettings).toContainEqual(channelParamToSetting[1][2]);
+      expect(channelSettings).toContainEqual(channelParamToSetting[2][2]);
+    }
+  });
+
+  it("overrides ch settings when per-channel settings are included", async () => {
+    const queryString = "?ch=0&lut=1,2&c1=v:1,lut:4:5";
+    const params = new URLSearchParams(queryString);
+    const { args } = await getArgsFromParams(params);
+
+    const groups = args.viewerChannelSettings?.groups[0]!;
+    expect(groups.channels).toHaveLength(1);
+    const channelSetting = groups.channels[0];
+
+    expect(channelSetting.match).toEqual(1);
+    expect(channelSetting.enabled).toEqual(true);
+    expect(channelSetting.lut).toEqual(["4", "5"]);
+  });
+
+  it("creates empty default data for bad per-channel setting formats", async () => {
```

Comment:
I can add a test for this, but if this were to happen it should only have defined settings for c0 and c2 as an output? This would make c1 use the default viewer settings, where both the isosurface and volume are not visible.